synapseclient/core/exceptions.py

```diff
@@ -15,6 +15,10 @@ class SynapseFileNotFoundError(SynapseError):
     """Error thrown when a local file is not found in Synapse."""
 
 
+class SynapseNotFoundError(SynapseError):
```

Comment:
Nit: does python base exceptions have something for Not found errors?

Reply:
I did not find one - The closet from: https://docs.python.org/3/library/exceptions.html

were:

1. FileNotFoundError - Derived from OSError and not applicable in all cases
2. ModuleNotFoundError - Not really the same thing